tests/unit/providers/common/outbound_filter_tests.py

```diff
@@ -0,0 +1,113 @@
+    outbound_user_filter,
+    outbound_user_creation_filter,
+)
+
+
+def test_outbound_user_filter():
+    """ Test outbound user filter with valid user """
+    result = outbound_user_filter({"user_id": 1234}, "azure")
+    assert isinstance(result, dict) is True
+    assert result["id"] == 1234
+    assert "job_title" not in result
+
+
+def test_outbound_user_filter_bad_provider():
+    """ Test outbound user filter with bad provider throws error"""
+    with pytest.raises(TypeError):
+        outbound_user_filter({"user_id": 1234}, "test_run")
+
+
+def test_outbound_group_filter():
+    """ Test outbound group filter with valid user """
+    result = outbound_group_filter({"role_id": 1234}, "ldap")
+    assert result["objectGUID"] == 1234
+    assert "id" not in result
+
+
+def test_outbound_group_filter_bad_provider():
+    """ Test outbound group filter with bad provider throws error"""
+    with pytest.raises(TypeError):
+        outbound_group_filter({"user_id": 1234}, "test_run")
+
+
+@pytest.mark.parametrize(
+    "test_input, provider, field_to_test, expected",
+    [
+        (
+            {
+                "account_enabled": True,
+                "name": "test_user",
+                "user_principal_name": "test_user@test_mail.com",
+            },
+            "azure",
+            "mailNickname",
+            "test_user",
+        ),
+        (
+            {"name": "test_user", "user_principal_name": "test_user@test_mail.com"},
+            "azure",
+            "accountEnabled",
+            True,
+        ),
+        (
+            {"name": "test_user", "user_principal_name": "test_user@test_mail.com"},
+            "ldap",
+            "accountEnabled",
+            True,
+        ),
+    ],
+)
+def test_outbound_user_creation(test_input, provider, field_to_test, expected):
+    """ Test outbound user creation with valid provider and account"""
+    assert (
+        outbound_user_creation_filter(test_input, provider)[field_to_test] == expected
+    )
+
+
+@pytest.mark.parametrize(
+    "test_input, provider, errorType",
+    [
+        ({"account_enabled": True}, "azure", ValueError),
+        (
+            {"name": "test_user", "user_principal_name": "test_user@test_mail.com"},
+            "Potate",
+            TypeError,
+        ),
+    ],
+)
+def test_outbound_user_creation_with_bad_input(test_input, provider, errorType):
+    """ Test outbound group creation with bad provider throws error"""
+    with pytest.raises(errorType):
+        outbound_user_creation_filter(test_input, provider)
+
+
+def test_outbound_group_creation():
+    """ Test outbound group creation with valid provider and group"""
+    user = {"group_nickname": "test_group"}
+    result = outbound_group_creation_filter(user, "azure")
+    assert result["mailEnabled"] is False
+    assert result["mailNickname"] == "test_group"
+    assert "mail" not in result
+
+
+@pytest.mark.parametrize(
+    "test_input, provider, errorType",
+    [
```

Comment:
In terms of parameterization - this works, but refer back to how I did it on azure_provider.py Instead of loading your multiple tests cases in duplication, you declare it once at the top and load it in as the declared name.

Reply:
Done!